src/main.rs

```diff
@@ -147,7 +152,17 @@ fn try_handle_admin<'a>(
             let mut found = None;
             for m in modules {
                 if m.name() == module {
-                    found = match m.admin(&mut *store, rest.trim(), sender) {
+                    // If the next word resolves to a valid room id use that, otherwise use the
+                    // current room.
+                    let (target_room, rest) = match rest.split_once(" ").map(|(l, r)| (l, r.trim()))
+                    {
+                        Some((possible_room, rest)) => match util.resolve_room(&possible_room) {
+                            Ok(resolved_room) => (resolved_room, rest.to_string()),
+                            Err(_) => (room.to_string(), format!("{} {}", possible_room, rest)),
+                        },
+                        None => (room.to_string(), rest.to_string()),
+                    };
```

Comment:
This could be hoisted out of the loop, since it's invariant per module (and that would avoid repeating the same work a few times).